includes/server/class-llms-rest-memberships-controller.php

```diff
@@ -0,0 +1,590 @@
+		if ( ! empty( $schema['properties']['restriction_action'] ) && isset( $request['restriction_action'] ) ) {
+			$prepared_item['restriction_redirect_type'] = $request['restriction_action'];
+		}
+
+		// Restriction message.
+		if ( ! empty( $schema['properties']['restriction_message'] ) && isset( $request['restriction_message'] ) ) {
+			if ( is_string( $request['restriction_message'] ) ) {
+				$prepared_item['restriction_notice'] = $request['restriction_message'];
+			} elseif ( isset( $request['restriction_message']['raw'] ) ) {
+				$prepared_item['restriction_notice'] = $request['restriction_message']['raw'];
+			}
+		}
+
+		// Restriction page id.
+		if ( ! empty( $schema['properties']['restriction_page_id'] ) && isset( $request['restriction_page_id'] ) ) {
+			$page = get_post( $request['restriction_page_id'] );
+			if ( $page && is_a( $page, 'WP_Post' ) ) {
+				$prepared_item['redirect_page_id'] = $request['restriction_page_id'];
+			} else {
+				$prepared_item['redirect_page_id'] = 0;
+			}
+		}
+
+		// Restriction URL.
+		if ( ! empty( $schema['properties']['restriction_url'] ) && isset( $request['restriction_url'] ) ) {
+			$prepared_item['redirect_custom_url'] = $request['restriction_url'];
+		}
+
+		// Sales page id.
+		if ( ! empty( $schema['properties']['sales_page_page_id'] ) && isset( $request['sales_page_page_id'] ) ) {
+			$page = get_post( $request['sales_page_page_id'] );
+			if ( $page && is_a( $page, 'WP_Post' ) ) {
+				$prepared_item['sales_page_content_page_id'] = $request['sales_page_page_id'];
+			} else {
+				$prepared_item['sales_page_content_page_id'] = 0;
+			}
+		}
+
+		// Sales page type.
+		if ( ! empty( $schema['properties']['sales_page_type'] ) && isset( $request['sales_page_type'] ) ) {
+			$prepared_item['sales_page_content_type'] = $request['sales_page_type'];
+		}
+
+		// Sales page URL.
+		if ( ! empty( $schema['properties']['sales_page_url'] ) && isset( $request['sales_page_url'] ) ) {
+			$prepared_item['sales_page_content_url'] = $request['sales_page_url'];
+		}
+
+		return $prepared_item;
+
+	}
+
+	/**
+	 * Prepare links for the request.
+	 *
+	 * @param LLMS_Membership $membership LLMS Membership.
+	 * @return array Links for the given object.
+	 */
+	protected function prepare_links( $membership ) {
+		$parent_links = parent::prepare_links( $membership );
+		$id           = $membership->get( 'id' );
+
+		$links = array();
+
+		// Enrollments.
+		$links['enrollments'] = array(
+			'href' => rest_url( sprintf( '/%s/%s/%d/%s', $this->namespace, $this->rest_base, $id, 'enrollments' ) ),
+		);
+
+		// Instructors.
+		$links['instructors'] = array(
+			'href' => add_query_arg(
+				'post',
+				$id,
+				rest_url( sprintf( '%s/%s', 'llms/v1', 'instructors' ) )
+			),
+		);
+
+		// Students.
+		$links['students'] = array(
+			'href' => add_query_arg(
+				'enrolled_in',
+				$id,
+				rest_url( sprintf( '%s/%s', 'llms/v1', 'students' ) )
+			),
+		);
+
+		return array_merge( $parent_links, $links );
```

Comment:
Could you please add a filter here?
https://github.com/eri-trabiccolo/lifterlms-rest/blob/memberships/includes/server/class-llms-rest-memberships-controller.php#L648

See #85 (comment)
Basically since then I've planned to add filters/action hooks to the other controllers too.